### ISSUE TITLE
Fix memory leaks in TextFormatter tests

### ISSUE THREAD

**Issue by Minoru** [newsboat/newsboat#622]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: c01acf47142637b8959036585bc30eaf28c30052

Steps to reproduce the issue:

1. Compile Newsboat's test suite with Clang's AddressSanitizer:

```
$ export CXX=clang++
$ export CXXFLAGS='-fsanitize=address -fno-omit-frame-pointer'
$ make test
```

2. Run `TextFormatter` tests: `(cd test && ./test '[TextFormatter]'`

3. Observe that AddressSanitizer reports some memory leaks:

```
Filters: [TextFormatter]
===============================================================================
All tests passed (22 assertions in 8 test cases)


=================================================================
==6278==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 256 byte(s) in 1 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aad0319 in regcomp (/lib/x86_64-linux-gnu/libc.so.6+0xde319)

Direct leak of 224 byte(s) in 1 object(s) allocated from:
    #0 0x51afb2 in realloc (/home/minoru/src/newsboat/test/test+0x51afb2)
    #1 0x7f935aacf97c  (/lib/x86_64-linux-gnu/libc.so.6+0xdd97c)

Indirect leak of 12288 byte(s) in 6 object(s) allocated from:
    #0 0x51ad8a in calloc (/home/minoru/src/newsboat/test/test+0x51ad8a)
    #1 0x7f935aac52ea  (/lib/x86_64-linux-gnu/libc.so.6+0xd32ea)
    #2 0x7f935aacb0af  (/lib/x86_64-linux-gnu/libc.so.6+0xd90af)

Indirect leak of 2048 byte(s) in 1 object(s) allocated from:
    #0 0x51ad8a in calloc (/home/minoru/src/newsboat/test/test+0x51ad8a)
    #1 0x7f935aac5470  (/lib/x86_64-linux-gnu/libc.so.6+0xd3470)
    #2 0x7f935aacb0af  (/lib/x86_64-linux-gnu/libc.so.6+0xd90af)

Indirect leak of 528 byte(s) in 6 object(s) allocated from:
    #0 0x51ad8a in calloc (/home/minoru/src/newsboat/test/test+0x51ad8a)
    #1 0x7f935aac3fae  (/lib/x86_64-linux-gnu/libc.so.6+0xd1fae)
    #2 0x7f935aacb0af  (/lib/x86_64-linux-gnu/libc.so.6+0xd90af)

Indirect leak of 128 byte(s) in 1 object(s) allocated from:
    #0 0x51ad8a in calloc (/home/minoru/src/newsboat/test/test+0x51ad8a)
    #1 0x7f935aacee43  (/lib/x86_64-linux-gnu/libc.so.6+0xdce43)

Indirect leak of 112 byte(s) in 1 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aacf208  (/lib/x86_64-linux-gnu/libc.so.6+0xdd208)

Indirect leak of 112 byte(s) in 1 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aacf1f7  (/lib/x86_64-linux-gnu/libc.so.6+0xdd1f7)

Indirect leak of 112 byte(s) in 1 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aacee0e  (/lib/x86_64-linux-gnu/libc.so.6+0xdce0e)

Indirect leak of 112 byte(s) in 7 object(s) allocated from:
    #0 0x51afb2 in realloc (/home/minoru/src/newsboat/test/test+0x51afb2)
    #1 0x7f935aac1821  (/lib/x86_64-linux-gnu/libc.so.6+0xcf821)

Indirect leak of 88 byte(s) in 1 object(s) allocated from:
    #0 0x51ad8a in calloc (/home/minoru/src/newsboat/test/test+0x51ad8a)
    #1 0x7f935aac3fae  (/lib/x86_64-linux-gnu/libc.so.6+0xd1fae)

Indirect leak of 28 byte(s) in 7 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aac1775  (/lib/x86_64-linux-gnu/libc.so.6+0xcf775)

Indirect leak of 28 byte(s) in 1 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aacf1d9  (/lib/x86_64-linux-gnu/libc.so.6+0xdd1d9)

Indirect leak of 28 byte(s) in 7 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aac3ce6  (/lib/x86_64-linux-gnu/libc.so.6+0xd1ce6)

Indirect leak of 28 byte(s) in 7 object(s) allocated from:
    #0 0x51ab93 in __interceptor_malloc (/home/minoru/src/newsboat/test/test+0x51ab93)
    #1 0x7f935aac3969  (/lib/x86_64-linux-gnu/libc.so.6+0xd1969)

SUMMARY: AddressSanitizer: 16120 byte(s) leaked in 49 allocation(s).
```

**Comment by Minoru** [comment]
Apparently this is fixed by #637. I'll check again when that PR is merged.

**Comment by Minoru** [comment]
This can't be reproduced on current master, 2e696502ba5d6109b6257ac33c24cd777eaab7f4, so it's fixed.
